cli/src/cli_util.rs

```diff
@@ -2732,28 +2742,47 @@ fn resolve_aliases(
                         r#"Recursive alias definition involving "{alias_name}""#
                     )));
                 }
-                if let Some(value) = aliases_map.remove(&alias_name) {
-                    if let Ok(alias_definition) = value.try_deserialize::<Vec<String>>() {
-                        assert!(string_args.ends_with(&alias_args));
-                        string_args.truncate(string_args.len() - 1 - alias_args.len());
-                        string_args.extend(alias_definition);
-                        string_args.extend_from_slice(&alias_args);
-                        resolved_aliases.insert(alias_name.clone());
-                        continue;
-                    } else {
-                        return Err(user_error(format!(
-                            r#"Alias definition for "{alias_name}" must be a string list"#
-                        )));
-                    }
+                if let Some(alias_definition) = resolved_alias_map.get(&alias_name) {
+                    assert!(string_args.ends_with(&alias_args));
+                    string_args.truncate(string_args.len() - 1 - alias_args.len());
+                    string_args.extend(alias_definition.clone());
+                    string_args.extend_from_slice(&alias_args);
+                    resolved_aliases.insert(alias_name.clone());
+                    continue;
                 } else {
                     // Not a real command and not an alias, so return what we've resolved so far
                     return Ok(string_args);
                 }
             }
         }
         // No more alias commands, or hit unknown option
-        return Ok(string_args);
+        break;
     }
+
+    // Add each alias to `Command` so it shows up in help
+    for (alias_name, alias_definition) in resolved_alias_map {
```

Comment:
Perhaps, this can be extracted to separate function of `(&config::Config, &Command) -> Command` type. That will help track which `app` object contains fake alias subcommands.

Downside is that we'll need to extract `aliases` map twice, but I think that's okay.

Reply:
oh, this is a good idea, thank you :) i separated this into a new `add_alias_help` function and changed `expand_args` to return the alias map so it doesn't need to parse it twice.